index.js

```diff
@@ -908,6 +908,12 @@ class HtmlWebpackPlugin {
    * The webpack base resolution path for relative paths e.g. process.cwd()
    */
   getFullTemplatePath (template, context) {
+    if (template === 'auto') {
+      template = path.resolve(context, 'src/index.ejs');
```

Comment:
I wouldn't override the value coming from the argument, I think is confusing in terms code readability. I suggest doing:

```js
getFullTemplatePath (template, context) {
  if (template === 'auto') {
    let templatePath = path.resolve(context, 'src/index.ejs');
    if (!fs.existsSync(templatePath)) {
     return path.join(__dirname, 'default_index.ejs');
  }
}
```

and later conusme the value return from the method, that way makes the function pure :)

Reply:
Thanks for your feedback 👍
We can't return early here because of the prefixing and path normalisation.

Reply:
Oh ok got it, was pretty sure I was missing context from the rest of the code. But would be cool to refactor in the future. Thanks for the explanation.